File: 0881-boats-to-save-people/0881-boats-to-save-people.py
class Solution:
  def numRescueBoats(self, people: List[int], limit: int) -> int:
    l = len(people)
    people.sort()
    l, r, b = 0, l-1, 0
    while l <= r:
      if people[l] + people[r] <= limit:
        l += 1
        r -= 1
      else:
        r -= 1
      b += 1
    return b

File: 0881-boats-to-save-people/0881-boats-to-save-people.cpp
class Solution {
public:
  int numRescueBoats(vector<int>& people, int limit) {
    int l = 0;
    int r = people.size()-1;
    int b = 0;
    sort(people.begin(), people.end());
    while (l <= r) {
      if (people[l] + people[r] <= limit) {
        ++l;
      }
      --r;
      ++b;
    }
    return b;
  }
};